src/main/java/com/edmunds/tools/databricks/maven/BaseDatabricksUpsertClusterMojoSettingsInitializer.java

```diff
@@ -39,11 +40,22 @@ public void fillInDefaults(ClusterSettingsDTO settingsDTO, ClusterSettingsDTO de
             log.info(String.format("set CusterName with %s", clusterName));
         }
 
-        int numWorkers = settingsDTO.getNumWorkers();
-        if (numWorkers == 0) {
-            numWorkers = defaultSettingsDTO.getNumWorkers();
-            settingsDTO.setNumWorkers(numWorkers);
-            log.info(String.format("%s|set NumWorkers with %s", clusterName, numWorkers));
+        AutoScaleDTO autoScale = settingsDTO.getAutoScale();
+        boolean autoScaleValid = autoScale != null && autoScale.getMinWorkers() != 0 && autoScale.getMaxWorkers() != 0
+                && autoScale.getMaxWorkers() >= autoScale.getMinWorkers();
+        if (!autoScaleValid) {
```

Comment:
this doesn't look right that we are checking twice if !autoScaleValid?

Reply:
Agree, it has been already simplified along with another changes (now waiting for a new databricks-rest-client version to commit)

Reply:
ok but it was intended, got it. approving